src/cal/mod.rs

```diff
@@ -0,0 +1,176 @@
+use std::collections::BTreeSet;
+use std::ops::Range;
+
+use chrono::DateTime;
+use chrono::Duration;
+use chrono::Utc;
+
+use std::cmp::Ordering;
+
+pub struct Cal {
+    events: BTreeSet<CmpEvent>,
+}
+
+#[allow(dead_code)]
+impl Cal {
+    pub fn new() -> Cal {
+        Cal {
+            events: BTreeSet::new(),
+        }
+    }
+
+    pub fn events_in(&self, range: Range<DateTime<Utc>>) -> impl Iterator<Item = Event> + '_ {
```

Comment:
`impl Iterator<Item = Event> + '_` doesn't tell callers anything about the iterator's lifetime. You should make clear the fact that the iterator's lifetime is tied to `self`'s. I.e.

`pub fn events_in<'a>(&'a self, range: Range<DateTime<Utc>>) -> impl Iterator<Item = Event> + 'a`